### PR TITLE
POC/K8s: Custom CRD Storage

### PR DESCRIPTION
**Pull request by toddtreece**
In this version of the POC, we are exploring implementing k8s api-server with a custom storage engine.  Initially backed by SQL.  See https://github.com/grafana/grafana/pull/66219 for another approach

`conf/custom.ini`:
```ini
app_mode = development 

[feature_toggles]
k8s = true
entityStore = true ; resources are saved in SQL tables
```

### CODE REVIEW

**Review by ryantxu — CHANGES_REQUESTED** [review]
requesting change... just to block merge

File: conf/k8s-authn-webhook-config [hunk 1]
```diff
@@ -0,0 +1,17 @@
+clusters:
+- name: authn-webhook
```
Comment (ryantxu) [inline conf/k8s-authn-webhook-config:2]: @charandas for the custom storage solution, we can remove the authnz webhooks correct?

or maybe just the authz one 🤔

File: pkg/api/http_server.go [hunk 2]
```diff
@@ -408,48 +428,85 @@ func (hs *HTTPServer) Run(ctx context.Context) error {
...
+		}()
 	default:
 		panic(fmt.Sprintf("Unhandled protocol %q", hs.Cfg.Protocol))
 	}
 
-	wg.Wait()
+	return nil
+}
+
+func (hs *HTTPServer) k8sWebhookServer() error {
```
Comment (ryantxu) [inline pkg/api/http_server.go:469]: IIUC, with custom storage we do not need a webhook server at all

File: conf/k8s-authn-webhook-config (hunk 1 above)
Comment (charandas) [reply conf/k8s-authn-webhook-config:2]: Yes, we can remove the `authz` webhook.

**Comment by toddtreece** [comment]
replaced by https://github.com/grafana/grafana/pull/67945

**Pull request closed without merging**